test/unittest/Database/DatabaseStatusTests.cpp

```diff
@@ -41,7 +41,9 @@ class database_status_tests : public ::testing::Test
         locator = db.locators().begin()->second;
 
         // Simulate that the backend is monitorizing the domain
+        // NOTE: This is so F*** dangerous, please do not do it again (1)
         std::shared_ptr<details::Monitor> monitor = std::make_shared<details::Monitor>();
+        monitor->id = domain->id;
         details::StatisticsBackendData::get_instance()->monitors_by_entity_[domain->id] = monitor;
```

Comment:
Since `Monitor` is an internal class, we can change the constructor to accommodate our needs. So I would have a constructor receiving the id, and then change this to:

```suggestion
        auto monitor = std::make_shared<details::Monitor>(domain->id);
        details::StatisticsBackendData::get_instance()->monitors_by_entity_[domain->id] = monitor;
```

Reply:
I think this is related with refactoring Monitor to be a class and not a struct. As this should be done at some point, I would not do this for the moment

Reply:
For Monitor refactor along with other needed refactors